src/drawing/DrawingAnnotations.tsx

```diff
@@ -54,6 +55,8 @@ const DrawingAnnotations = (props: Props): JSX.Element => {
     const hasDrawnPathGroups = drawnPathGroups.length > 0;
     const hasStashedPathGroups = stashedPathGroups.length > 0;
     const hasPathGroups = hasDrawnPathGroups || hasStashedPathGroups;
+    const popupDrawingToolbarRef = React.useRef<PopupBaseRef>(null);
+    const prevDrawnPathGroups = usePrevious(drawnPathGroups);
```

Comment:
Added this because we were calling update too many times - update should only be called once when drawnPathGroups changes.